File: problems/starterpackages/Tron.java
import java.util.*;
import java.io.*;

public class Tron {

	//We can't print to console, because that channel is used for IO with the environment.
	//We therefore create a PrintWriter and output to that. Use "OutputDebug" rather than "System.out.print".
	public static PrintWriter debug;
	public static <T> void outputDebug(T s) {
		if(debug == null) {
			try {
				debug = new PrintWriter("debug-" + (System.currentTimeMillis() / 1000) + ".log"); //Cast to seconds.
			} catch(Exception e) {
				System.out.println("Debug file could not be open.");
				System.exit(1);
			}
		}
		debug.print(s);
		debug.flush();
	}

	public static enum Direction {
		NORTH, EAST, SOUTH, WEST
	}

	public static enum Tile {
		EMPTY, PLAYER1, PLAYER2, 
		TAKEN_BY_PLAYER1, TAKEN_BY_PLAYER2
	}

	private static int[][] deserializeMap(String mapString) {
		String[] tiles = mapString.trim().split(" ");
		int[] tileVals = new int[tiles.length];
		for (int i=0; i<tiles.length; i++) tileVals[i] = Integer.parseInt(tiles[i].trim());

		int[][] map = new int[16][16];
		for (int i=0; i<16; i++) map[i] = Arrays.copyOfRange(tileVals,i*16,(i+1)*16);
		return map;
	}

	private static String getString() {
		try {
			StringBuilder builder = new StringBuilder();
			int buffer;
			while ((buffer = System.in.read()) >= 0) {
				if (buffer == '\n') {
					break;
				} else {
					builder = builder.append((char)buffer);
				}
			}
			return builder.toString();
		} catch(Exception e) {
			System.exit(1);
			return null;
		}
    }

	public static int[][] getMap() {
		String message = getString().trim();
		if(message.equals("KILL")) {
			outputDebug("exit");
			System.exit(0);
		}
		return deserializeMap(message);
	}

	public static void sendMove(Direction move) {
		System.out.print(move.ordinal()+"\n");
		System.out.flush();
	}
}

File: problems/starterpackages/Tron.py
import sys
import time

north = 0
east = 1
south = 2
west = 3

empty = 0
myCar = 1
opponentCar = 2
takenByMe = 3
takenByOpponent = 4

def log(string):
	with open("debug.log", "a") as debug:
		debug.write(str(string)+"\n")

class Networker:
	def deserializeMap(self, mapString):
		tiles = [int(tile) for tile in mapString.split(" ")]
		map = [tiles[a*16 : (a+1)*16] for a in range(16)]
		return map

	def getMap(self):
		message = sys.stdin.readline().strip()
		if message == "KILL": sys.exit()
		return self.deserializeMap(message)

	def sendMove(self, move):
		sys.stdout.write(str(move) + "\n")
		sys.stdout.flush()